app/lib/index.js

```diff
@@ -221,6 +216,34 @@ renderer.on('config:set', function(key, value, ...args) {
   }
 });
 
+// error tracking //////////
+renderer.on('sentry:initialize', function(params) {
```

Comment:
To understand this one better: Why do we need to wait for the client to initialize before initializing sentry, server side?

I'd assume we initialize sentry as quick as possible.

Reply:
Because initializing Sentry depends on Privacy Preferences - Enable Error Tracking option. We don't want to pollute backend code with business logic. Backend should wait until the ErrorTracking decides if we should initialize Sentry or not (after reading privacy preferences)

Reply:
I think we should read the existing privacy preferences as early as possible and initialize Sentry ASAP. Otherwise, we are never going to catch crashes before the client runs, even if user allows crash reports.

Reply:
It sounds better on the paper but that'd mean we introduce a lot of business logic into the backend code. Do we really want to read "Privacy Preferences" which is under the scope of a client plugin within backend?

As Maciej mentioned we'd then also catch errors earlier (before the plugins are initialized), how ever **IMO** it does not worth it to introduce Privacy Preferneces stuff into app bootrsaping code. I'd rather keep things under their related scopes (ErrorTracking plugin should be decisive if we'll initialize Sentry or not)

Reply:
On the other hand, I think that catching crashes on start of application *is* our business logic.

Reply:
So if we want to go to that direction then PrivacyPreferences should not be a plugin at all.

**Why?**

Because the whole idea of plugins is to have something pluggable in order to help us strucured/pluggable code, easy to insert and remove later on.

What if we are to deprecate PrivacyPreferences plugin? I'd expect I just remove the folder and that's it, I no longer have PrivacyPreferences. If we're to have privacy preferences stuff inside backend code, it's not the case anymore. It feels like 🍝  code.

Reply:
Should we maybe then have privacy prefs as a global utility or something?

Reply:
Another condition is also: DISABLE_REMOTE_INTERACTION flag. We'd then have to add another check for this flag in backend code. I'd rather have them all under the plugin code.

Reply:
How is error tracking dependent on "privacy preferences" as a full feature? 

We'd need to have a small util in the backend that is aware of both the DISABLE_* as well as the privacy prefs flag but that would be all. All these things can safely be read from the config.

Reply:
I would rather say that the "plugin" part is more like "UI for changing privacy preferences" while the privacy preferences as such are in the very core of the application. Also, as @nikku pointed, the backend part can be built as a small utility. It does not have to know about the client, as config is read and written on the backend.

Reply:
Then we could also use the official Sentry Electron API since we're ok with checking privacy preferences in backend now.